### PR TITLE
Add analysis for concept exercise `marios-marvellous-lasagna`

### LINKED ISSUE #14: Write analysis for concept exercise `marios-marvellous-lasagna`
[From `design.md`](https://github.com/exercism/elm/blob/main/exercises/concept/marios-marvellous-lasagna/.meta/design.md):

This exercise could benefit from the following rules added to the the analyzer:
- make sure that `let` is used somewhere

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #14.

[Sister PR here](https://github.com/exercism/website-copy/pull/2227).

I only added one rule for checking if a `let` has been used (I added the possibility in `Analyzer.functionCalls`).

Is it worth adding a rule checking that there are two expressions defined in the let block or what their names are? It wouldn't be very hard, but there isn't that much benefit, unless we want to make sure that they know that you can also define functions in there, not just values.

### CODE REVIEW

File: src/Analyzer.elm [hunk 1]
```diff
@@ -141,6 +144,10 @@ initialContext functions =
         |> Rule.withModuleNameLookupTable
 
 
+
+-- Keep track of top level function
```
Comment (ceddlyburge) [inline src/Analyzer.elm:148]: Can we use the documentation function comment style that elm-format likes here?

```{-| .... -}```

File: src/Analyzer.elm [hunk 2]
```diff
@@ -166,36 +173,45 @@ annotateFunctionDeclaration calledFrom node (Context context) =
             ( [], Context context )
 
 
+
+-- Gather relevant expressions from the code
```
Comment (ceddlyburge) [inline src/Analyzer.elm:177]: Can we use the documentation function comment style that elm-format likes here?

```{-| .... -}```

File: src/Exercise/MariosMarvellousLasagna.elm [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ruleConfig : RuleConfig
+ruleConfig =
+    { slug = Just "marios-marvellous-lasagna"
+    , restrictToFiles = Just [ "src/MariosMarvellousLasagna.elm" ]
+    , rules = [ CustomRule usesLet ]
+    }
+
+
+
+{- There should be a `let` use somewhere -}
```
Comment (ceddlyburge) [inline src/Exercise/MariosMarvellousLasagna.elm:19]: Can we use the documentation function comment style that elm-format likes here?

```{-| .... -}```

File: test_data/two-fer/imperfect_solution/expected_analysis.json [hunk 4]
```diff
@@ -1 +1 @@
...
...onable","params":{"definition":"5| -- unused imported module\n6| import Parser.Advanced\n          ^^^^^^^^^^^^^^^"}},{"comment":"elm.common.simplify","type":"actionable","params":{"message":"Simplify: Unnecessary multiplication by 1\n\n34|     -- something is unused\n35|     Just something -> 1 * 1\n                           ^^^^\n36|     -- True && False can be simplified to False\n\nMultiplying by 1 does not change the value of the number."}},{"comment":"elm.common.simplify","type":"actionable","params":{"message":"Simplify: Part of the expression is unnecessary\n\n37|     -- if condition can be simplified to 0\n38|     Nothing -> if (True && False) then 1 else 0\n                       ^^^^^^^^^^^^^\n\nA part of this condition is unnecessary. You can remove it and it would not impact the behavior of the program."}},{"comment":"elm.common.simplify","type":"actionable","params":{"message":"Simplify: Comparison is always False\n\n37|     -- if condition can be simplified to 0\n38|     Nothing -> if (True && False) then 1 else 0\n                       ^^^^^^^^^^^^^\n\nThis condition will always result in the same value. You may have hardcoded a value or mistyped a condition."}}]}
```
Comment (ceddlyburge) [inline test_data/two-fer/imperfect_solution/expected_analysis.json:1]: Is this due to the update in elm-review?

**Review by ceddlyburge — APPROVED** [review]

File: test_data/two-fer/imperfect_solution/expected_analysis.json (hunk 4 above)
Comment (jiegillet) [reply test_data/two-fer/imperfect_solution/expected_analysis.json:1]: Yes! 
Some messages changed the wording a bit, and the order had changed for some reason (but I can't be mad, it's not like the order is guaranteed by the elm-review API) so I added a secondary order to our comments to help with that. 
I suppose approach would be to use `jq` to order the fields before comparing them but that might be a bit tricky too.

File: src/Analyzer.elm (hunk 1 above)
Comment (jiegillet) [reply src/Analyzer.elm:148]: Ah, I never noticed it did that

**Comment by jiegillet** [comment]
All done!

@exercism/maintainers-admin  could we get an approval?

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Pull request merged**